api/krusty/options.go

```diff
@@ -36,6 +36,10 @@ type Options struct {
 	// When true, use kyaml/ packages to manipulate KRM yaml.
 	// When false, use k8sdeps/ instead (uses k8s.io/api* packages).
 	UseKyaml bool
+
+	// When true, allow name and kind changing via a patch
+	// When false, patch name/kind don't overwrite target name/kind
+	ChangeId bool
```

Comment:
AllowResourceIdChanges

Reply:
Also, connect this to a command line flag to the  `build` command
 - see how this is done with enable_kyaml